crates/iceberg-rest-server/src/api/v1/warehouse.rs

```diff
@@ -75,6 +105,113 @@ pub trait WarehouseService<C: Catalog, A: AuthZHandler, S: SecretStore> {
             warehouse_id: warehouse_id.into_uuid(),
         })
     }
+
+    async fn update_warehouse(
+        request: UpdateWarehouseRequest,
+        context: ApiContext<State<A, C, S>>,
+        _headers: HeaderMap,
+    ) -> Result<UpdateWarehouseResponse> {
+        let warehouse_id = &WarehouseIdent::from(request.warehouse_id);
+        let mut transaction = C::Transaction::begin_write(context.v1_state.catalog.clone()).await?;
+        let old_warehouse_config =
+            C::get_warehouse(warehouse_id, context.v1_state.catalog.clone()).await?;
+        let (old_name, old_storage_profile, old_storage_credential_id) = {
+            (
+                old_warehouse_config.warehouse_name,
+                old_warehouse_config.storage_profile,
+                old_warehouse_config.storage_credential_id,
+            )
+        };
+
+        if let Some(new_name) = request.new_name {
+            if new_name != old_name {
+                C::update_warehouse_name(warehouse_id, &new_name, transaction.transaction())
+                    .await?;
+            }
+        }
+
+        match (request.new_storage_credential, request.new_storage_profile) {
+            (None, Some(_)) => {
+                let err: IcebergErrorResponse = ErrorModel::builder()
```

Comment:
There are some Storage Profiles that don't need credentials. Sorry, I forgot that case earlier.
In this case of the match statement we should validate the Storage Profile just like you did above, if it works, delete the old secret and store the new profile with storage_credential_id = None